workflows_api/runtime/src/airflow_helpers.py

```diff
@@ -0,0 +1,160 @@
+            "Authorization": "Bearer " + mwaa_cli_token["CliToken"],
+            "Content-Type": "application/json",
+        },
+        data=cli_string,
+    )
+    if mwaa_response.raise_for_status():
+        raise Exception(
+            f"Failed to trigger airflow: {mwaa_response.status_code} "
+            f"{mwaa_response.text}"
+        )
+    else:
+        return WorkflowExecutionResponse(**mwaa_response)
+
+
+def trigger_discover(input: Dict) -> Dict:
+    if not (MWAA_ENV := os.environ.get("MWAA_ENV")):
+        raise HTTPException(status_code=400, detail="MWAA environment not set")
+
+    airflow_client = boto3.client("mwaa")
+    mwaa_cli_token = airflow_client.create_cli_token(Name=MWAA_ENV)
+
+    mwaa_webserver_hostname = (
+        f"https://{mwaa_cli_token['WebServerHostname']}/aws_mwaa/cli"
+    )
+
+    unique_key = str(uuid4())
+    raw_data = f"dags trigger veda_discover --conf '{input.json()}' -r {unique_key}"
+    mwaa_response = requests.post(
+        mwaa_webserver_hostname,
+        headers={
+            "Authorization": "Bearer " + mwaa_cli_token["CliToken"],
+            "Content-Type": "application/json",
+        },
+        data=raw_data,
+    )
+    if mwaa_response.raise_for_status():
+        raise Exception(
+            f"Failed to trigger airflow: {mwaa_response.status_code} "
+            f"{mwaa_response.text}"
+        )
+    else:
+        return WorkflowExecutionResponse(
+            **{
+                "id": unique_key,
+                "status": Status.started,
+            }
+        )
+
+
+def list_dags() -> str:
+    if not (MWAA_ENV := os.environ.get("MWAA_ENV")):
+        raise HTTPException(status_code=400, detail="MWAA environment not set")
+
+    airflow_client = boto3.client("mwaa")
+    mwaa_cli_token = airflow_client.create_cli_token(Name=MWAA_ENV)
+
+    mwaa_webserver_hostname = (
+        f"https://{mwaa_cli_token['WebServerHostname']}/aws_mwaa/cli"
+    )
+
+    raw_data = "dags list"
+    mwaa_response = requests.post(
+        mwaa_webserver_hostname,
+        headers={
+            "Authorization": "Bearer " + mwaa_cli_token["CliToken"],
+            "Content-Type": "application/json",
+        },
+        data=raw_data,
+    )
+    if mwaa_response.raise_for_status():
+        raise Exception(
+            f"Failed to trigger airflow: {mwaa_response.status_code} "
+            f"{mwaa_response.text}"
+        )
+    else:
+        return mwaa_response.text
+
+
+def get_status(dag_run_id: str) -> Dict:
+    """
+    Get the status of a workflow execution.
+    """
+    if not (MWAA_ENV := os.environ.get("MWAA_ENV")):
+        raise HTTPException(status_code=400, detail="MWAA environment not set")
+
+    airflow_client = boto3.client("mwaa")
+    mwaa_cli_token = airflow_client.create_cli_token(Name=MWAA_ENV)
+
+    mwaa_webserver_hostname = (
+        f"https://{mwaa_cli_token['WebServerHostname']}/aws_mwaa/cli"
+    )
+
+    raw_data = "dags list-runs -d veda_discover"
```

Comment:
As in: should `Get the status of a workflow execution` read `Get the status of a veda_discover workflow execution by dag_id` to describe the current get_status method? I don’t fully understand what is happening in that function but it looks like all runs of the veda discover dag are retrieved and then searched to find the given dag run id?

I need to get more familiar with the mwaa system so I don't know if this is a common pattern that I am just seeing for the first time :)

Reply:
I agree with the doc change - I also think it would be helpful to be able to specify other DAGs, but I don't think we need to worry about it for this PR.